src/main/kotlin/com/gumi/moodle/rest_controllers/GradeController.kt

```diff
@@ -93,12 +94,29 @@ fun Application.gradeRoutes() {
                                 Course::grades.posOp / Grade::studentPoints atKey studentID setTo grade
                             ) { Course::_id eq it withGradeID gradeID }
 
-                            if (updated) call.respond(HttpStatusCode.OK)
-                            else call.respond(HttpStatusCode.NotModified)
+                            if (updated) {
+                                createNotification(userDao, courseID, gradeID, studentID)
+                                call.respond(HttpStatusCode.OK)
+                            } else call.respond(HttpStatusCode.NotModified)
                         }
                     }
                 }
             }
         }
     }
-}
\ No newline at end of file
+}
+
+private suspend fun createNotification(userDao: UserDAO, courseID: String, gradeID: String, studentID: String) {
+    val notification = Notification(courseID, gradeID, System.currentTimeMillis())
+
+    userDao.updateOne(
+        studentID,
+        pullByFilter(User::notifications, (Notification::courseID eq courseID) and (Notification::gradeID eq gradeID))
+    ) { User::_id eq it }
+
+    userDao.updateOne(
+        studentID,
+        push(User::notifications, notification)
+    ) { User::_id eq it }
+}
```

Comment:
Czy nie lepiej tę funkcję dać do utils?

Reply:
Utils pasują do rzeczy które mają zastosowanie uniwersalne, ale zgadzam się, że nie warto psuć elegancji klas od endpointów, najlepiej byłoby zrobić osobną klasę od tego, ale trochę szkoda dla jednej metody, to przeniosę do utils.